platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java

```diff
@@ -500,18 +502,23 @@ private static boolean includeModuleLanguageLevelOverride(Project rootProject, I
     }
 
     private void configureForScalaPlugin() {
+        boolean isolatedProjects = getBuildFeatures().getIsolatedProjects().getActive().get();
         project.getPlugins().withType(ScalaBasePlugin.class, new Action<ScalaBasePlugin>() {
             @Override
             public void execute(ScalaBasePlugin scalaBasePlugin) {
-                ideaModuleDependsOnRoot();
+                ideaModuleDependsOnRoot(isolatedProjects);
             }
         });
         if (isRoot()) {
-            new IdeaScalaConfigurer(project).configure();
+            new IdeaScalaConfigurer(project, isolatedProjects).configure();
```

Comment:
💭 Instead of passing the flag, passing a callback would be nicer as it would allow the `isolatedProjects` dependent code to be centralized on this class

Reply:
Wrapping things in actions makes the classes less concrete / harder to reason about, which is something I would think twice about in the Gradle codebase. But in this case, it indeed seems good to consolidate the failure mode, especially because it's temporary.